core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -909,8 +911,35 @@ private[group] class GroupCoordinator(
         val group = groupManager.getGroup(groupId).getOrElse {
           groupManager.addGroup(new GroupMetadata(groupId, Empty, time))
         }
-        doTxnCommitOffsets(group, transactionalId, memberId, groupInstanceId, generationId, producerId, producerEpoch,
-          offsetMetadata, requestLocal, responseCallback)
+
+        val offsetTopicPartition = new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, partitionFor(group.groupId))
+
+        def postVerificationCallback(
+          error: Errors,
+          newRequestLocal: RequestLocal,
+          verificationGuard: VerificationGuard
+        ): Unit = {
+          if (error != Errors.NONE) {
+            val finalError = error match {
+              case Errors.NOT_ENOUGH_REPLICAS => Errors.COORDINATOR_NOT_AVAILABLE
```

Comment:
Why is only this error is converted? It looks like NOT_LEADER is another possibility prior to verification? Would it make sense to factor out the conversion logic in `createPutCacheCallback` so that we are less likely to miss any errors?

Reply:
createPutCacheCallback is not called here. 

There are three places 

1 -> replica manager calls not leader before we get to verification
2 -> on verification we convert those errors to known produce errors
3 -> in group coordinator we convert any produce errors not known to the offset api

You can see I added the path for not leader prior to verification. I kept that as is (also added a test)
If we think that should not be returned via the offset commit, we can add another case. I just thought I wanted to convert the verification specific errors in a common place and then group coordinator specific errors in the group coordinator code. I also didn't think it makes sense to call createPutCacheCallback since we can just return the error response directly.

Reply:
Synced offline -- I made a method for the conversions in group coordinator and share among the putCacheCallback and verification callback. I also added the tests for the errors.